tuist/tuist
# Check the latest version from Google Cloud Storage

Related https://github.com/tuist/tuist/issues/848

### Short description 📝
As explained [in this issue](https://github.com/tuist/tuist/issues/848) we are moving away from GitHub releases to Google Cloud Storage as a place to store our releases. Several users ran into issues in their projects because they were hitting some limits on the GitHub API.

### Solution 📦
This is the first of 2 PRs to make this change possible. It adds a new utility, `GoogleCloudStorageClient` to interact with Google Cloud Storage, and uses it from the `Updater` utility to check what's the latest version available.

In a follow-up PR, I'll extend `GoogleCloudStorageClient` to support downloading a given version and update the `Updater` accordingly. After that users should no longer run into API limit issues.

### How to test
Check out the branch and run:

```
rm -rf ~/.tuist/Versions
swift run tuistenv update
```

tuistenv should be able to determine the latest version and update to it.
